R/ops-addition.r

```diff
@@ -17,11 +17,10 @@ add_duration_to_date <- function(dur, date) {
     date <- as.POSIXct(date)
     ans <- with_tz(date + dur@.Data, "UTC")
 
-    if (all(is.na(ans))) return(as.Date(ans))  # ALL NAs
-
-    if (all(hour(na.omit(ans)) == 0  &
-            minute(na.omit(ans)) == 0 &
-            second(na.omit(ans)) == 0)) {
+    if (all(is.na(ans) |
```

Comment:
This logic doesn't seem quite right to me — I think it would be better to decide to return a date or not depending on whether the duration is a integer multiple of 86400 or not. Then I think this logic would get a lot simpler.